ca/ca.go

```diff
@@ -52,14 +52,36 @@ type issuerMaps struct {
 	byNameID map[issuance.NameID]*issuance.Issuer
 }
 
+// DefaultCertProfileName will be assigned to any configured certificate profile
+// without a Name.
+const DefaultCertProfileName = "defaultBoulderCertificateProfile"
```

Comment:
This should only be defined in //cmd/boulder-ca/main.go, and even there it should only be in the form of `if c.CA.Issuance.DefaultCertificateProfileName == "" { c.CA.Issuance.DefaultCertificateProfileName = "defaultBoulderCertificateProfile" }`.

From then on, the configured default value needs to be carried forward as a field either on the CAImpl struct, or a field on the certProfilesMaps struct.

Right now, all uses of the default below just use this value directly, meaning that the default from the config file is ignored entirely. I believe this bug would have been caught if the integration test config specified a default other than "defaultBoulderCertificateProfile".